lib/usbmux/index.js

```diff
@@ -41,12 +41,21 @@ function swap16 (val) {
   return ((val & 0xFF) << 8) | ((val >> 8) & 0xFF);
 }
 
+/**
+ * @typedef {Object} SocketOptions
+ * @property {?string} socketPath [/var/run/usbmuxd] The full path to the usbmuxd Unix socket
```

Comment:
Just as a fyi, it may make sense to allow the caller to force TCP mode, even when on Linux/macOS. This can be useful, for example, when connecting to muxer on a remote host (for example, if usbmuxd and Appium are running in a different pod on a Kubernetes cluster, or, in more general terms, if the device is connected to a remote server).

For example, we could fall back to TCP mode if `socketPath` is null, or respect the [`USBMUXD_SOCKET_ADDRESS`](https://github.com/libimobiledevice/libusbmuxd/blob/13de0244fdf20ba1abc7361140f159d00d9067dd/README.md#environment) environment variable like [libusbmuxd](https://github.com/libimobiledevice/libusbmuxd/) does.

I don't have an immediate use case for this now, but it came to mind, and thought it might be useful to share.

Reply:
Thanks for the info. To be honest, currently we don't really develop this module further, so we'd be happy to accept any improvements/fixes to it.